scientific_library/tvb/simulator/simulator.py

```diff
@@ -203,6 +212,7 @@ def _configure_integrator_boundaries(self):
         else:
             self.integrator.bounded_state_variable_indices = None
             self.integrator.state_variable_boundaries = None
+        self.bound_and_clamp = self.integrator.bound_and_clamp
```

Comment:
Why keep the `bound_and_clamp` on the simulator?

Reply:
So, I removed the bound_and_clamp form the simulator. Using the one of integrator, and applying it to the history as well by default.
As for the two distinct methods for integrate() and integrate_with_update() they are necessary to avoid an if statement for every time step of integration, I guess.